plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt

```diff
@@ -219,13 +224,55 @@ fun reauthConnectionIfNeeded(
     if (tokenProvider is InteractiveBearerTokenProvider) {
         onPendingToken(tokenProvider)
     }
-    return reauthProviderIfNeeded(project, tokenProvider)
+    return reauthProviderIfNeeded(project, tokenProvider, connection)
 }
 
-fun reauthProviderIfNeeded(project: Project?, tokenProvider: BearerTokenProvider): BearerTokenProvider {
+fun reauthProviderIfNeeded(
```

Comment:
```suggestion
private fun reauthProviderIfNeeded(
```

not really a fan of requiring the connection to be passed with the provider, since one is derived from the other